### DISCUSSION TITLE
Question: Multiline Button

### DISCUSSION THREAD

**Discussion by graysonhicks** [segmentio/evergreen#1258] [Q&A]
Love this component library.  Quick question:

The Button seems to keep the text on a single line, even when the height is increased.  What's the best way to wrap the text?

Thanks!

**Comment by jeroenransijn** [comment]
Hey @graysonhicks thanks for checking out Evergreen and kicking the tires. Could you give a little bit more details on the result you are trying to achieve — maybe a screenshot or napkin sketch works. There might be some ways to hack around this shortcoming, otherwise we'll have to figure out if we want to support that use case.

Thanks!

**Comment by graysonhicks** [comment]
Sure, no problem. (This is just a coding exercise so it's not a deal-breaker, but I do really like Evergreen and thought I would ask about it).

 Basically I have some buttons with some very long text.  The text overflows the button, and the button stays on one line and does not grow in heigh to accommodate the text.

Here's a screenshot:

<img width="495" alt="screen shot 2018-04-29 at 12 28 11 am" src="https://user-images.githubusercontent.com/16200795/39428815-bfe26e44-4c56-11e8-8d2c-de4e7e062a30.png">

Expected behavior: button grows in height to fit both lines of text.

**Comment by jeroenransijn** [comment]
Thanks for sharing your use case. Seems like something we might want to support. It will be tricky to support this without adding a new API or creating a new component sadly. This is because a lot of the components such as the buttons are controlled by a fixed height.

**Comment by jeroenransijn** [comment]
I would recommend moving some content out of the button and changing the label to something like `Update State`

**Comment by graysonhicks** [comment]
I understand, thanks.  Would also note though, that this would seem pretty common on small screen widths with a lot fewer words.  

I may mess around to see if there is some simple styling that would allow a button to grow with text with a special prop.

**Comment by jeroenransijn** [comment]
This has become an issue for as well! Looping in @vadimdemedes here who is experiencing this issue in our app. I think we probably need to patch this before v4.

![image 5](https://user-images.githubusercontent.com/564463/40745811-5390f372-640d-11e8-8264-46c49bd7677e.png)
